### ISSUE TITLE
VariableForProperty should not fail if modifier is used

### ISSUE THREAD

**Issue by Anahkiasen** [causes/scss-lint#375]
Example:

```scss
background-color: darken($color-primary, 15);
```

This to me is perfectly valid: your color is still abstracted into a variable and you're keeping the relationship with it by applying a modifier on it. But currently this fails VariableForProperty, what do others think, do you think this is correct or should this not failed/be allowed via an option?

**Comment by sds** [comment]
Hey @Anahkiasen,

What you're describing is adding additional complexity for a specific use case. 

You've already made an explicit decision to opt-in to using `VariableForProperty` for the `background-color` property. Adding special cases to recognize what you would call "modifiers" would introduce what I believe is unnecessary complexity.

By making exceptions for `darken`, `lighten` etc. you are going against the spirit of the linter, as you are still "redefining" colors. You should create `$color-primary-darker` and `$color-primary-lighter` variables so your code remains DRY.

**Comment by Anahkiasen** [comment]
Yeah I guess, valid point. I think my issue with VariableForProperty overall is that it doesn't take into account how many times a color is used. I have a lot of "one off" colors where it would actually be counterproductive to put them in variables.

I think the VariableForProperty should have a configurable threshold of how many times a value is used before it is recommended to be put into a variable. Like "Ok you used `color: black` once through the entire app, you can leave that" but "Ok you used `#CCCCCC` 5 times in all your stylesheets, would be best to put it in a variable". Would that be more sensible?

**Comment by lencioni** [comment]
That sounds great to me in theory, but I don't think it will work with scss-lint because it would require scss-lint to make a full pass on all of your `.scss` files every time it runs, which is not how it operates.
